Dockerfile

```diff
@@ -1,10 +1,23 @@
-FROM ruby:3.1
+ARG RUBY_VERSION=3.1
+ARG NODE_MAJOR=19
 
-# Ensure node.js 19 is available for apt-get
-RUN curl -sL https://deb.nodesource.com/setup_19.x | bash -
+FROM ruby:$RUBY_VERSION-slim-bullseye
 
 # Install dependencies
-RUN apt-get update -qq && apt-get install -y build-essential libvips nodejs && npm install -g yarn
+RUN apt-get update -qq \
+  && DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+    build-essential \
+    libvips \
+    gnupg2 \
+    curl \
+  && apt-get clean
+
+RUN curl -sL https://deb.nodesource.com/setup_$NODE_MAJOR.x | bash -
+RUN apt-get update -qq && DEBIAN_FRONTEND=noninteractive apt-get -yq dist-upgrade && \
+  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+    nodejs \
+    && npm install -g yarn \
+    && apt-get clean
```

Comment:
How about using multistage build to pull in binaries without having to install node, like it was proposed in https://github.com/rails/docked/pull/13?

Reply:
I understand the advantages, but I think it's better the way we do it. But if everyone sees it better 🙂